onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -1,109 +1,40 @@
-
-    # loop through all the images in a directory
-    for image in os.listdir(image_directory):
-        image_file = os.path.join(image_directory, image)
-
-        #call the batch file to process the image[i] in the location and generate output log files
-        # --> TODO <--
-        preprocess_images_with_debug_op(image_file)
-    
-        # open each log file and read through it line by line and ignore 'labels' file 
-        for filename in os.listdir(logs_directory):
-            if filename.endswith(".txt") and (filename.startswith("labels.txt") == False): 
-                with open(os.path.join(logs_directory, filename), 'r') as file_object:
-                    line = file_object.readline()
-                    spaces_removed_data = []
-                    while line:
-                        # at each line check for a match with a regex
-                        key, match = _parse_line(line)
-                        if key == 'data': 
-                            data = match.group(0)
-                            for i in data.split(','):
-                                try:
-                                    if i != '':
-                                        spaces_removed_data.append(float(i.strip()))
-                                except ValueError:
-                                    print("error on line"+ i)
-
-                            #add min and max values to the dictionary
-                            if (filename.split(".")[0] not in maxvalue.keys()) or (max(spaces_removed_data) > maxvalue[filename.split(".")[0]]):
-                                maxvalue[filename.split(".")[0]] = max(spaces_removed_data)
-                            if (filename.split(".")[0] not in minvalue.keys()) or (min(spaces_removed_data) < minvalue[filename.split(".")[0]]):
-                                minvalue[filename.split(".")[0]] = min(spaces_removed_data)
-                    
-                        line = file_object.readline() #next line in file
-
-                    file_object.close() #close the file object
-    
-    #clean up the log files to repeat steps for the next image
-    cleanup_images_from_debug_op_run()
-
-    #print(json.dumps(maxvalue, indent = 4)) #dump the max dictionary
-    #print(json.dumps(minvalue, indent = 4)) #dump the min dictionary
-    for conv_output in maxvalue:
-        scale = (maxvalue[conv_output] - minvalue[conv_output]) / 255 if minvalue[conv_output] !=maxvalue[conv_output] else 1
-        zero_point = round((0 - minvalue[conv_output]) / scale)
-        print(f"{conv_output} : {scale} , {zero_point}")
-
-        # Replace by this
-        # quantized_model = quantize(model, per_channel=False, nbits=8, quantization_mode=QuantizationMode.QLinearOps_Dynamic,
-# 		asymmetric_input_types=False, input_quantization_params=_input_quantization_params,
-# 		output_quantization_params=_output_quantization_params)
+from onnx import helper, TensorProto
+
+def augment_graph(model):
```

Comment:
Don't replace the original script... original script did not have graph augmentation... you are adding this capability so make your changes as additions... When you do the next step to run inference on this augmented model that's when you should replace the corresponding functionality in the script

Reply:
is there a better name than augment_graph() ?  from api/naming perspective, it's rather vague.

Reply:
augment_graph is the first step towards calibration. Users will call Calibrate API (which will be added soon) This higher level calibrate api will call augment graph as first step and then call collect quantization threasholds (all this will be added soon)

Reply:
if it's only used internally, maybe it's ok.
But in the PR description, with the example usage I got the impression it's a public api on the same level as onnx.load and onnx.save

Reply:
Yes, augment_graph will function as a helper method within this script.